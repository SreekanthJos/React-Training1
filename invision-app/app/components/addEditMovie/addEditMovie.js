import React from 'react';
import './addMovie.scss';
import { useFormik } from 'formik';
import PropTypes from 'prop-types';

const validate = (values) => {
  const errors = {};
  for (let key in values) {
    if (!values[key]) {
      errors[key] = 'error';
    }
    if (key === 'poster_path') {
      // eslint-disable-next-line no-useless-escape
      const regexp = /(ftp|http|https):\/\/(\w+:{0,1}\w*@)?(\S+)(:[0-9]+)?(\/|\/([\w#!:.?+=&%@!\-\/]))?/;
      if (!regexp.test(values[key])) {
        errors[key] = 'error';
      }
    }
    if (key === 'runtime' && isNaN(values[key])) {
      errors[key] = 'error';
    }
  }
  return errors;
};


export function AddEditMovie(props) {
  const {handleSubmit,handleChange,resetForm,values,errors} = useFormik({
    initialValues: {
      id: props?.movie?.id || Math.floor(Math.random() + Date.now()),
      title: props?.movie?.title,
      release_date: props?.movie?.release_date,
      poster_path: props?.movie?.poster_path,
      genres: props?.movie?.genres,
      overview: props?.movie?.overview,
      runtime: props?.movie?.runtime
    },
    validate,
    onSubmit:  (values)=> {
      event.preventDefault();
      props.onSubmit(values);     
      props.close();
    }
  });
  
  return (
    <div className='add-modal-dialog'>
      <div className='add-modal-dialog-content'>
        <a href='#close' title='close' className='close' onClick={props.close}>X</a>
        <h2 className='add-modal-dialog-header'></h2>
        <form className='add-modal-dialog-form' onSubmit={handleSubmit}>
          {props.movie &&
            <label>
              Movie Id
        <input name='id' readOnly className='add-input title' type="text"  value={values.id} />
            </label>
          }
          <label>
            Title
        <input name='title' className={`add-input title ${ errors.title }`} type="text" placeholder='Title' onChange={handleChange} value={values.title || ''} />
          </label>
          <label>
            Release date
        <input name='release_date' className={`add-input title ${ errors.release_date }`} type="date" onChange={handleChange} value={values.release_date || ''} />
          </label>
          <label>
            Movie url
        <input name='poster_path' className={`add-input title ${ errors.poster_path }`} type="text" placeholder='Movie URL here' onChange={handleChange} value={values.poster_path || ''} />
          </label>
          <label>
            Genre
        <select name='genres' className={`add-select title ${ errors.genres }`} onChange={handleChange} value={values.genres || ''}>
              <option value="selectTitle">Select Genre</option>
              <option value="documentary">Documentary</option>
              <option value="comedy">Comedy</option>
              <option value="horror">Horror</option>
              <option value="crime">Crime</option>
            </select>
          </label>
          <label>
            Overview
        <input name='overview' className={`add-input title ${ errors.overview }`} type="text" placeholder='Overview here' onChange={handleChange} value={values.overview || ''} />
          </label>
          <label>
            Runtime
        <input name='runtime' className={`add-input title ${ errors.runtime }`} type="text" placeholder='Runtime here' onChange={handleChange} value={values.runtime || ''} />
          </label>
          <div className='button-container'>
            <input className='reset-button' type='reset' value='Reset' onClick={resetForm} />
            <input className='submit-button' type='submit' value='Submit' />
          </div>
        </form>
      </div>
    </div>
  );
}

AddEditMovie.propTypes = {
  movie: PropTypes.object,
  onSubmit: PropTypes.func.isRequired, 
  close: PropTypes.func.isRequired
};

































