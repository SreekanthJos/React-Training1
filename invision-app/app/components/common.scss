
.logo{
    color:#f65261;
    font-size: 18px;
}

  .logo__title {
    font-family: 'Aller-light';

  }

  .logo__titlebold {
    font-family: 'Aller-bold';
  }

.header {
    position: absolute;
    right: 0;
    top: 0;
    left: 0;
    z-index: 1;
    display: flex;
    align-items: center;
    height: 48px;
    padding-left: 24px;
    background: transparent;
    text-align: left;
  }

  .footer {
    position: fixed;
    right: 0;
    bottom: 0;
    left: 0;
    z-index: 1;
    display: flex;
    align-items: center;
    justify-content: center;
    height: 48px;
    background: transparent;
    color: #f65261;
    text-align: center;
  }
  
.button {
  padding: 10px 20px;
  border: none;
  border-radius: 4px;
  outline: none;
  font-size: 16px; 
  cursor: pointer;
}
 
  .button__control {
    background-color: #424242;
    color: #f65261;
  }

  .button__primary {
    padding: 16px 40px;
        background-color: #f65261;
        color: #fff;
  }