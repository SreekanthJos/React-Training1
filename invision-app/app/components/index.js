export * from './Header'
export * from './Footer'
export * from './Home'
export * from './addMovie'
export * from './search'
export * from './movieFilter'
export * from './movieFiterAndSort'
export * from './sortBy'
export * from './moviesFound'
export * from './movieCard'
export * from './movieList'