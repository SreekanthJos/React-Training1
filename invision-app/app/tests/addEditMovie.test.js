
import React from 'react';
import TestRenderer from 'react-test-renderer';
import { AddEditMovie } from '../components/addEditMovie';
import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store';
const mockStore = configureStore();
const store = mockStore({});

let props = {  
  close: jest.fn(),
  onSubmit:jest.fn()
};

describe('Add Edit Movie popup:', () => {
  test('should render Add Edit Movie popup', () => {
    const searchBar = TestRenderer.create(<Provider store={store}><AddEditMovie {...props}/></Provider>);
    const json = searchBar.toJSON();
    expect(json).toMatchSnapshot();
  })
});