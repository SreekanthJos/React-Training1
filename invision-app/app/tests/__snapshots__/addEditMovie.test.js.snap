// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Add Edit Movie popup: should render Add Edit Movie popup 1`] = `
<div
  className="add-modal-dialog"
>
  <div
    className="add-modal-dialog-content"
  >
    <a
      className="close"
      href="#close"
      onClick={[MockFunction]}
      title="close"
    >
      X
    </a>
    <h2
      className="add-modal-dialog-header"
    >
      Add Movie
    </h2>
    <form
      className="add-modal-dialog-form"
      onSubmit={[Function]}
    >
      <label>
        Title
        <input
          className="add-input title undefined"
          name="title"
          onChange={[Function]}
          placeholder="Title"
          type="text"
          value=""
        />
      </label>
      <label>
        Release date
        <input
          className="add-input title undefined"
          name="release_date"
          onChange={[Function]}
          type="date"
          value=""
        />
      </label>
      <label>
        Movie url
        <input
          className="add-input title undefined"
          name="poster_path"
          onChange={[Function]}
          placeholder="Movie URL here"
          type="text"
          value=""
        />
      </label>
      <label>
        Genre
        <select
          className="add-select title undefined"
          name="genres"
          onChange={[Function]}
          value=""
        >
          <option
            value="selectTitle"
          >
            Select Genre
          </option>
          <option
            value="documentary"
          >
            Documentary
          </option>
          <option
            value="comedy"
          >
            Comedy
          </option>
          <option
            value="horror"
          >
            Horror
          </option>
          <option
            value="crime"
          >
            Crime
          </option>
        </select>
      </label>
      <label>
        Overview
        <input
          className="add-input title undefined"
          name="overview"
          onChange={[Function]}
          placeholder="Overview here"
          type="text"
          value=""
        />
      </label>
      <label>
        Runtime
        <input
          className="add-input title undefined"
          name="runtime"
          onChange={[Function]}
          placeholder="Runtime here"
          type="text"
          value=""
        />
      </label>
      <div
        className="button-container"
      >
        <input
          className="reset-button"
          onClick={[Function]}
          type="reset"
          value="Reset"
        />
        <input
          className="submit-button"
          type="submit"
          value="Submit"
        />
      </div>
    </form>
  </div>
</div>
`;
